### ISSUE TITLE
[a11y] Radio buttons are keyboard inaccessible 

### ISSUE THREAD

**Issue by lesliexin** [Shopify/polaris-react#3570]
## Issue summary

[Edit] This seems to be an issue with radio buttons in general, not only inside a modal

When[ radio buttons](https://polaris.shopify.com/components/forms/radio-button) are present inside a [modal](https://polaris.shopify.com/components/overlays/modal), they are inaccessible through tabbing and cannot be triggered through a key press.

See [Slack](https://shopify.slack.com/archives/CQK6FNLJK/p1603823046068100) discussion for more context.
[Code Sandbox](https://codesandbox.io/s/thirsty-edison-x02l0?file=/App.js)

![image](https://user-images.githubusercontent.com/18248358/97357344-fc675900-186f-11eb-8be6-5efe4c47ce63.png)

## Expected behavior

- When the modal is opened, a user should be able to navigate through and focus on each radio button option with the tab key
- Once the desired option is in focus, the user should be able to check the radio button using the enter key
 

## Actual behavior

See screen recording: https://screenshot.click/2020-10-27_0kf7h-t77iu-4n565.mp4

## Steps to reproduce the problem

1. Locate a modal with radio buttons
- plus admin users -> security -> global TFA card
- plus admin users -> security -> SAML card
- /admin/settings/checkout
2. Attempt to navigate the radio button options with only the keyboard

## Reduced test case

Code sandbox: https://codesandbox.io/s/thirsty-edison-x02l0?file=/App.js

## Specifications

- Are you using the React components? (Y/N):
- Polaris version number:
- Browser:
- Device:
- Operating System:

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```
System:
    OS: macOS 10.15.5
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 939.00 MB / 32.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.22.1 - ~/.nvm/versions/node/v10.22.1/bin/node
    Yarn: 1.22.0 - ~/.dev/yarn/1.22.0/bin/yarn
    npm: 6.14.6 - ~/.nvm/versions/node/v10.22.1/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 86.0.4240.111
    Safari: 13.1.1
  npmPackages:
    @shopify/polaris: 5.9.1 => 5.9.1
    react: ~16.11.0 => 16.11.0
    react-dom: ~16.11.0 => 16.11.0

```

**Comment by thayannevls** [comment]
Hello @lesliexin 👋  I checked the [WAI-ARIA documentation](https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/radio/radio.html) and is described that the radio group should be navigated by the `up` and `down` arrow keys. The radio group is working just fine with these arrow keys, you can check on this [screen recording](https://drive.google.com/file/d/1A3FZY4bTlCRaI9k9kZDQ0gk9Mn-jyjwW/view?usp=sharing) or testing yourself on the codesandbox you created here.

There is any particular necessity to the radio group be navigable by the `tab` and `enter` keys ? Or with just the arrow keys is enough?

If there is anything to change, I would be happy to contribute 😄

**Comment by lesliexin** [comment]
@thayannevls Hi there! Sorry for my late response :) After further discussion there seems to be no problem with the current behaviour of the component! I can close this issue :)
